client/src/home/add_button/AddButton.js

```diff
@@ -47,23 +49,44 @@ function AddItemForm(props) {
             <Form.Control
               type="text"
               placeholder="e.g. Apples"
-              onChange={(event) =>
-                setItemName(event.target.value.trim().toLowerCase())
-              }
+              onChange={(event) =>{
+                let val = event.target.value.trim().toLowerCase();
+                if(val.length > 0){
+                  setItemName(event.target.value.trim().toLowerCase());
+                  setitemNameWarning(false);
+                }
+                else{
+                  setitemNameWarning(true);
+                }
+                
+              }}
             />
+            {showitemNameWarning && (
+              <Form.Text className="warning">Item name is required</Form.Text>
+            )}
           </Form.Group>
-          
-          <b>OR</b>
 
           <Form.Group>
-            <Form.Label>Item Number</Form.Label>
+            <Form.Label>Batch ID</Form.Label>
             <Form.Control
               type="text"
               placeholder="e.g. 1234"
               onChange={(event) => {
-                setItemId(event.target.value.trim().toLowerCase());
+                let val = event.target.value.trim().toLowerCase();
+                if(val.length > 0){
+                  setbatchId(event.target.value.trim().toLowerCase());
```

Comment:
Same issue here. The batchID state should always reflect what is currently in the field even if the field is empty.

Reply:
Fixed!